src/pages/CovidHistory.js

```diff
@@ -105,17 +114,38 @@ const CovidHistory = ({
                   max={moment().format(DATE_FORMAT.yyyymmdd)}
                   disabled={
                     `${history.field}` === "dateOfDose1Vaccination"
-                      ? !checkedOne
-                      : !checkedTwo
+                      ? !intakeState.covidVaccinationDose1Taken
```

Comment:
please move all this stuff to a function, then you can set `disabled={getIsDisabled(history?.field)}`